app/models/identifier.rb

```diff
@@ -27,21 +27,27 @@ class Identifier < ActiveRecord::Base
 
   belongs_to :identifiable, polymorphic: true
 
-  belongs_to :identifier_scheme
+  # TODO: uncomment 'optional: true' once we are on Rails 5
+  belongs_to :identifier_scheme #, optional: true
 
   # ===============
   # = Validations =
   # ===============
 
-  validates :identifier_scheme,
-            presence: { message: PRESENCE_MESSAGE },
-            uniqueness: { scope: %i[identifiable_id identifiable_type],
-                          message: UNIQUENESS_MESSAGE }
-
   validates :value, presence: { message: PRESENCE_MESSAGE }
 
   validates :identifiable, presence: { message: PRESENCE_MESSAGE }
 
+  validate :value_uniqueness_with_scheme, if: :has_scheme?
+
+  validate :value_uniqueness_without_scheme, unless: :has_scheme?
+
+  # =============
+  # = Callbacks =
+  # =============
+
+  before_save :append_scheme_prefix
```

Comment:
Would overriding the default accessors work here like you'd done in the other ror-fundref PR?  I liked how that could be tested by changing the value on an instanciated record rather than needing to test the function that the callback calls.

https://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Overwriting+default+accessors

Reply:
yess absolutely. will change this